rocketpy/Dispersion.py

```diff
@@ -1065,490 +487,503 @@ def run_dispersion(
+                        name=fin,
+                    )
 
             # Tail
-            for tail in self.tail_names:
+            for tail in setting["tails"].keys():
                 rocket_dispersion.addTail(
-                    topRadius=setting[f"tail_{tail}_topRadius"],
-                    bottomRadius=setting[f"tail_{tail}_bottomRadius"],
-                    length=setting[f"tail_{tail}_length"],
-                    position=setting[f"tail_{tail}_position"],
-                    radius=None,
-                    name="Tail",
+                    length=setting["tails"][tail]["length"],
+                    position=setting["tails"][tail]["position"],
+                    topRadius=setting["tails"][tail]["topRadius"],
+                    bottomRadius=setting["tails"][tail]["bottomRadius"],
+                    # radius=setting["tails"][tail]["radius"],
+                    # TODO: understand if we need vary this radius argument
+                    name=tail,
                 )
 
+            # Rail Buttons
+            for rail_buttons in setting["rail_buttons"].keys():
+                rocket_dispersion.setRailButtons(
+                    position=[
+                        setting["rail_buttons"][rail_buttons]["upper_button_position"],
+                        setting["rail_buttons"][rail_buttons]["lower_button_position"],
+                    ],
+                    angular_position=setting["rail_buttons"][rail_buttons][
+                        "angular_position"
+                    ],
+                )
             # Add parachutes
-            rocket_dispersion.parachutes = []  # Remove existing parachutes
-            for name in self.parachute_names:
+            for chute in setting["parachutes"].keys():
                 rocket_dispersion.addParachute(
-                    name=name,
-                    CdS=setting["parachute_" + name + "_CdS"],
-                    trigger=setting["parachute_" + name + "_trigger"],
-                    samplingRate=setting["parachute_" + name + "_samplingRate"],
-                    lag=setting["parachute_" + name + "_lag"],
-                    noise=setting["parachute_" + name + "_noise"],
+                    name=chute,
+                    CdS=setting["parachutes"][chute]["CdS"],
+                    trigger=setting["parachutes"][chute]["trigger"],
+                    samplingRate=setting["parachutes"][chute]["samplingRate"],
+                    lag=setting["parachutes"][chute]["lag"],
+                    noise=setting["parachutes"][chute]["noise"],
                 )
 
-            rocket_dispersion.setRailButtons(
-                position=[
-                    setting["positionFirstRailButton"],
-                    setting["positionSecondRailButton"],
-                ],
-                angularPosition=setting["railButtonAngularPosition"],
-            )
+            # TODO: Fix rail buttons definition here
```

Comment:
@MateusStano , we still need to fix rail button in this part of the code, right? I thought you had fixed it already

Reply:
Just forgot to remove the commented lines